code/game/gamemodes/crash.dm

```diff
@@ -0,0 +1,154 @@
+/datum/game_mode/crash
+	name = "Crash"
+	config_tag = "Crash"
+	required_players = 0
+	flags_round_type = MODE_INFESTATION|MODE_FOG_ACTIVATED
+	flags_landmarks = MODE_LANDMARK_SPAWN_XENO_TUNNELS|MODE_LANDMARK_SPAWN_MAP_ITEM
+
+	// Round start conditions
+	var/xeno_required_num = 1
+	var/xeno_starting_num = 0
+	var/list/xenomorphs = list()
+	
+	// Round end conditions
+	var/marines_evac = FALSE
+	var/planet_nuked = FALSE
+
+
+/datum/game_mode/crash/announce()
+	to_chat(world, "<span class='round_header'>The current map is - [SSmapping.config.map_name]!</span>")
+
+
+/datum/game_mode/crash/can_start()
+	. = ..()
+	// Check if enough players have signed up for xeno & queen roles.
+	initialize_xeno_leader()
+	initialize_xenomorphs()
+	if(!length(xenomorphs)) // we need at least 1
+		return FALSE
+
+	return TRUE
+
+
+/datum/game_mode/crash/pre_setup()
+	. = ..()
+
+	for(var/i in xenomorphs)
+		var/datum/mind/M = i
+		if(M.assigned_role == ROLE_XENO_QUEEN)
+			transform_queen(M)
+		else
+			transform_xeno(M)
+
+
+/datum/game_mode/crash/check_finished()
+	if(round_finished)
+		return TRUE
+
+	if(world.time < (SSticker.round_start_time + 15 SECONDS))
+		return FALSE
+
+	var/list/living_player_list = count_humans_and_xenos()
+	var/num_humans = living_player_list[1]
+	var/num_xenos = living_player_list[2]
+	
+	var/list/grounded_living_player_list = count_humans_and_xenos(SSmapping.levels_by_any_trait(list(ZTRAIT_GROUND)))
+	var/num_grounded_humans = grounded_living_player_list[1]
+	
+	if(!planet_nuked && num_humans == 0 && num_xenos > 0) // XENO Major (All marines killed)
+		message_admins("Round finished: [MODE_CRASH_X_MAJOR]")
+		round_finished = MODE_CRASH_X_MAJOR 
+	else if(marines_evac && planet_nuked && (num_humans == 0 || num_grounded_humans > 0)) // Marine minor (Planet nuked, no human left on planet)
+		message_admins("Round finished: [MODE_CRASH_M_MINOR]")
+		round_finished = MODE_CRASH_M_MINOR 
+	else if(marines_evac && planet_nuked) // Marine Major (Planet nuked, marines evac)
+		message_admins("Round finished: [MODE_CRASH_M_MAJOR]")
+		round_finished = MODE_CRASH_M_MAJOR 
+	else if(marines_evac && !planet_nuked) // Xeno minor (Marines evac, planet intact)
+		message_admins("Round finished: [MODE_CRASH_X_MINOR]")
+		round_finished = MODE_CRASH_X_MINOR 
+	else if(num_humans == 0 && num_xenos == 0) // Draw, for some unknown reason
+		message_admins("Round finished: [MODE_CRASH_DRAW_DEATH]")
+		round_finished = MODE_CRASH_DRAW_DEATH 
+
+	return FALSE
+
+/datum/game_mode/crash/declare_completion()
+	. = ..()
+	to_chat(world, "<span class='round_header'>|[round_finished]|</span>")
+	to_chat(world, "<span class='round_body'>Thus ends the story of the brave men and women of the [CONFIG_GET(string/ship_name)] and their struggle on [SSmapping.config.map_name].</span>")
+	
+	var/musical_track
+	switch(round_finished)
+		if(MODE_CRASH_X_MAJOR) // marines killed
```

Comment:
I think an assoc list would be better here for accessing the tracks, check POD_STYLES for inspiration